src/lightning_app/core/app.py

```diff
@@ -520,6 +526,84 @@ def state_dict(self) -> Dict:
     def load_state_dict(self, state: Dict) -> None:
         self.set_state(state)
 
+    @classmethod
+    def _load_checkpoint_from_json_file(cls, path: str) -> None:
+        checkpoint_dict = json.load(open(path))
+        # TODO: Checkpoint versioning and compatibility checks
```

Comment:
Where will this TODO be addressed?
- In this PR?
- Not in this PR? -> Let's open a GitHub issue

Reply:
created a GH issue for it: https://github.com/Lightning-AI/lightning/issues/15448